src/ops/checked.rs

```diff
@@ -90,3 +90,54 @@ checked_impl!(CheckedDiv, checked_div, i32);
 checked_impl!(CheckedDiv, checked_div, i64);
 checked_impl!(CheckedDiv, checked_div, isize);
 
+/// Performs a left shift that returns `None` on overflow.
+pub trait CheckedShl<RHS>: Sized + Shl<RHS, Output=Self> {
```

Comment:
I think for now, I'd rather keep these consistent with the other `Checked` traits that are not parameterized, especially since the standard library methods are only for `u32`.  That also moots your comment below about needing checked conversions to `u32`.

(We could decide to expand it later using a default, like `CheckedShl<RHS = u32>`.)